File: sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        def dfs(index, current_xor):
            # Base case: when all elements have been considered
            if index == len(nums):
                return current_xor
            # Include nums[index] in the subset
            include = dfs(index + 1, current_xor ^ nums[index])
            # Exclude nums[index] from the subset
            exclude = dfs(index + 1, current_xor)
            return include + exclude
        
        return dfs(0, 0)

File: sum-of-all-subset-xor-totals.java
class Solution {
    public static int subsetXORSum(int[] nums) {
        int total = 0;
        int n = nums.length;
        int subsets = (int)Math.pow(2, n); 

        for (int i = 0; i < subsets; i++) {
        	
            int xor = 0;
            
            for (int j = 0; j < n; j++) {

                int bit = (int)Math.pow(2, j);
                
                if ((i / bit) % 2 == 1) {
                    xor = xor ^ nums[j];
                }
                
            }
            
            total += xor;
        }

        return total;
    }
}
